File: app_codebase/dart_bkend_base_02/bin/modules/ws/demo_errors.dart
/// Module-owned WebSocket demo error codes.
library;

import '../../core/errors/contracts/register_module_error_contract.dart';
import '../../core/errors/error_spec.dart';

const demoRoomNotImplemented = ErrorSpec(
  'ws/demo_room/not_implemented',
  'Room subscribe not implemented',
  501,
);

void registerDemoErrors(ModuleErrorRegistrar registrar) {
  registrar.registerModule('ws', [demoRoomNotImplemented]);
}

File: app_codebase/python_base_05/bin/modules/ws/demo_errors.py
"""Module-owned WebSocket demo error codes."""

from core.errors.contracts.register_module_error_contract import ModuleErrorRegistrar
from core.errors.error_spec import ErrorSpec

DEMO_ROOM_NOT_IMPLEMENTED = ErrorSpec(
    "ws/demo_room/not_implemented",
    "Room subscribe not implemented",
    http_status=501,
)


def register_demo_errors(registrar: ModuleErrorRegistrar) -> None:
    registrar.register_module("ws", [DEMO_ROOM_NOT_IMPLEMENTED])
